src/js/internal/debugger.ts

```diff
@@ -264,30 +264,30 @@ function bufferedWriter(writer: Writer): Writer {
 const defaultHostname = "localhost";
 const defaultPort = 6499;
 
-function parseUrl(url: string): URL {
+export function parseUrl(url: string): URL {
   try {
     if (!url) {
       return new URL(randomId(), `ws://${defaultHostname}:${defaultPort}/`);
     } else if (url.startsWith("/")) {
       return new URL(url, `ws://${defaultHostname}:${defaultPort}/`);
-    } else if (/^[a-z+]+:\/\//i.test(url)) {
-      return new URL(url);
     } else if (/^\d+$/.test(url)) {
       return new URL(randomId(), `ws://${defaultHostname}:${url}/`);
-    } else if (!url.includes("/") && url.includes(":")) {
-      return new URL(randomId(), `ws://${url}/`);
-    } else if (!url.includes(":")) {
-      const [hostname, pathname] = url.split("/", 2);
-      return new URL(`ws://${hostname}:${defaultPort}/${pathname}`);
     } else {
-      return new URL(randomId(), `ws://${url}`);
+      const base = new URL(`ws://${url}`);
+      if (base.port === "") {
+        base.port = defaultPort.toString();
+      }
+      if (base.pathname !== "/") {
+        return base;
+      }
+      return new URL(randomId(), base);
     }
   } catch {
     throw new TypeError(`Invalid hostname or URL: '${url}'`);
   }
 }
 
-function randomId() {
+export function randomId() {
```

Comment:
Neither of these functions should be exported. `export` breaks the build here. These can only be `export default` at the very bottom of the file.

Reply:
Instead, these can be tested with an integration test, e.g. running `bun --inspect={}`, then connecting to the WebSocket and verifying the URL.